src/c2goto/headers/pthread.h

```diff
@@ -27,6 +27,10 @@
 #define NULL 0
 #endif
 
+/* Declaration of external variables */
+extern unsigned short int __ESBMC_num_threads_running;
+extern unsigned short int __ESBMC_blocked_threads_count;
```

Comment:
I'd suggest to not do that in the header file which is user-facing. You could declare these for instance in `semaphore_lib.c` instead or create a new "internal" header inside `c2goto/library`.

Reply:
Great, many thanks!